File: app/src/main/java/com/backyardev/spacejam/activities/SettingsActivity.java
package com.backyardev.spacejam.activities;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Toast;

import com.backyardev.spacejam.R;
import com.backyardev.spacejam.fragments.MainSettingsFragment;
import com.backyardev.spacejam.fragments.ProfileSettingFragment;

public class SettingsActivity extends AppCompatActivity {

    private static String TAG_PROFILE_SETTING="profileSettings";
    private static String TAG_MAIN_SETTING="mainSettings";
    private static String CURR_TAG=TAG_MAIN_SETTING;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate( savedInstanceState );
        setContentView( R.layout.activity_settings );
        setMainFragment();
    }

    private void setMainFragment() {
        {
            MainSettingsFragment fragment = new MainSettingsFragment();
            getSupportFragmentManager().beginTransaction().add( R.id.settings_container, fragment ).commit();
        }
        getIncomingIntent();
    }

    private void getIncomingIntent() {
        Intent intent = getIntent();
        if (intent.hasExtra( getString( R.string.calling_activity ) )) {
            getFragment(intent.getStringExtra( getString( R.string.calling_activity ) ));
        }
    }

    private void getFragment(String activity){
        if(activity.equals( "ProfileSettingsFragment" )){
            Toast.makeText( this,activity,Toast.LENGTH_SHORT ).show();
            ProfileSettingFragment profileSettingFragment = new ProfileSettingFragment();
            CURR_TAG=TAG_PROFILE_SETTING;
            setFragment( profileSettingFragment,CURR_TAG );
        }

    }
    

    private void setFragment(Fragment fragment, String TAG){
        FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
        fragmentTransaction.setCustomAnimations( android.R.anim.fade_in, android.R.anim.fade_out );
        fragmentTransaction.replace( R.id.settings_container, fragment, TAG );
        if (getSupportFragmentManager().findFragmentByTag( TAG ) == null) {
            fragmentTransaction.addToBackStack( TAG );
        }
        fragmentTransaction.commit();
    }
}

File: app/src/main/java/com/backyardev/spacejam/activities/SettingsActivity.kt
package com.backyardev.spacejam.activities

import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment

import android.os.Bundle

import com.backyardev.spacejam.R
import com.backyardev.spacejam.fragments.MainSettingsFragment
import com.backyardev.spacejam.fragments.ProfileSettingFragment

class SettingsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)
        setMainFragment()
    }

    private fun setMainFragment() {
        run {
            val fragment = MainSettingsFragment()
            supportFragmentManager.beginTransaction().add(R.id.settings_container, fragment).commit()
        }
        getIncomingIntent()
    }

    private fun getIncomingIntent() {
        val intent = intent
        if (intent.hasExtra(getString(R.string.calling_activity))) {
            getFragment(intent.getStringExtra(getString(R.string.calling_activity)))
        }
    }

    private fun getFragment(activity: String) {
        if (activity == "ProfileSettingsFragment") {
            val profileSettingFragment = ProfileSettingFragment()
            CURR_TAG = TAG_PROFILE_SETTING
            setFragment(profileSettingFragment, CURR_TAG)
        }
    }

    private fun setFragment(fragment: Fragment, TAG: String) {
        val fragmentTransaction = supportFragmentManager.beginTransaction()
        fragmentTransaction.setCustomAnimations(android.R.anim.fade_in, android.R.anim.fade_out)
        fragmentTransaction.replace(R.id.settings_container, fragment, TAG)
        if (supportFragmentManager.findFragmentByTag(TAG) == null) {
            fragmentTransaction.addToBackStack(TAG)
        }
        fragmentTransaction.commit()
    }

    companion object {
        private val TAG_PROFILE_SETTING = "profileSettings"
        private val TAG_MAIN_SETTING = "mainSettings"
        private var CURR_TAG = TAG_MAIN_SETTING
    }
}